src/main/java/com/karankumar/bookproject/ui/shelf/CustomShelfForm.java

```diff
@@ -154,6 +201,17 @@ private void addClassNameToForm() {
                             ".classList.add('shelfFormOverlay');\n");
     }
 
+    public void editShelf(String chosenShelf) {
+        delete.setVisible(true);
+        deletingShelf = chosenShelf;
+        shelfNameField.setValue(deletingShelf);
+        dialog.open();
+    }
+
+    public CustomShelf getCustomShelf(String shelfToFind) {
+        return customShelfService.findAll(deletingShelf).get(0);
```

Comment:
I guess you would want to use the `shelfToFind` in the method?

Reply:
Yep! That will be changed.